test/test_expand.jl

```diff
@@ -26,6 +26,9 @@ using Test
     @test isequal(qexpand((B+C+D)*A), B*A + C*A + D*A)
     @test isequal(qexpand(commutator(A, B) * C), A*B*C - B*A*C)
 
+    @test isequal(qexpand(A*(B⊗C⊗D)), (A*B)⊗(A*C)⊗(A*D))
+    @test isequal(qexpand((B⊗C⊗D)*A), (B*A)⊗(C*A)⊗(D*A))
+
     @test isequal(qexpand((A⊗B)*(C⊗D)), (A*C)⊗(B*D))
```

Comment:
I did not notice this originally, but this is only true if `basis(A)==basis(C)` and `basis(B)==basis(D)`. Could you add a test that confirms an error is raised when this is not the case?